features/crc/mbed_crc.h

```diff
@@ -0,0 +1,102 @@
+/* mbed Microcontroller Library
```

Comment:
location - `features/crc` ? this is not part of platform folder? - just thinking why feature/ location was chosen

Reply:
Current CRC use case is block device and filesystem, hence selected feature. 
In future mbed_crc.h class can be used as base class for hardware CRC, hence we can have it HAL layer as well. 
Inputs from core-team will be appreciated for this.

Reply:
I would assume it should go to drivers/ straight away.

Reply:
move entire CRC implementation to drivers? Or just mbed_crc.h.
Naming convention in driver folder does not have `mbed` prefix. Should I rename file to be CRC.h?
If everything is moved inside driver, should we have folder structure for software implementation or everything should be at same level?

Reply:
I would say all CRC files and we should match the convention for the drivers and I don't think we need directory structure for it.